sklearn/utils/estimator_checks.py

```diff
@@ -813,6 +797,24 @@ def check_estimator_sparse_data(name, estimator_orig, strict_mode=True):
                 else:
                     expected_probs_shape = (X.shape[0], 4)
                 assert probs.shape == expected_probs_shape
+        except (TypeError, ValueError) as e:
```

Comment:
I put back the old try/except here (instead of the newly introduced `with raises(...`), because I want to test first that if a `TypeError` or `ValueError` has been thrown, the right error msg is thrown, and then that if it's not this type of error but some general `Exception` that has been thrown, a `TypeError` or `ValueError` should be thrown, and I didn't manage to do this hierarchical testing with `with raises(...`